lib/apiservers/engine/backends/container.go

```diff
@@ -389,6 +382,55 @@ func (c *Container) ContainerExecStart(ctx context.Context, eid string, stdin io
 	actor := CreateContainerEventActorWithAttributes(vc, map[string]string{})
 	EventService().Log(event, eventtypes.ContainerEventType, actor)
 
+	ca := &backend.ContainerAttachConfig{
+		UseStdin:  ec.OpenStdin,
+		UseStdout: ec.OpenStdout,
+		UseStderr: ec.OpenStderr,
+	}
+
+	if ec.Tty {
+		ca.UseStderr = false
+	}
+
+	ac := &AttachConfig{
+		ID: eid,
+		ContainerAttachConfig: ca,
+		UseTty:                ec.Tty,
+		CloseStdin:            true,
+	}
+
+	err = c.containerProxy.AttachStreams(context.Background(), ac, stdin, stdout, stderr)
```

Comment:
Don't know if we should call this if all of the UseX are false (`-d` case). `AttachStreams` shuts down stdin on return which will kill the transport to the client. I don't know if we need that transport to return launch errors - my guess is we do.

Suggest we check for successful launch immediately after taskinspect (`ec.Running` for success/failure, `ec.ProcessConfig.ErrMessage` for detail) and only perform the attach prep if launch was successful.